hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -467,7 +486,7 @@ def add_line(self):
                     self.env['account.analytic.line'].create(values)
 
     def clean_timesheets(self, timesheet):
-        if self.id:
+        if self.id and self.state == 'draft':
```

Comment:
@astirpe 
`self.id` ??
Why check `self` ?, direct put 'if self.state == 'draft':`

Reply:
The check on `self` is needed. If you debug you can notice that `self` could be empty.